app/experimenter/experiments/models.py

```diff
@@ -241,6 +248,16 @@ def test_tube_url(self):
             ).format(slug=self.slug)
 
     def generate_normandy_slug(self):
+        if self.is_addon_study:
+            if not self.addon_experiment_id:
+                raise ValueError(
+                    (
+                        "An Add-On experiment requires a unique Active "
```

Comment:
```suggestion
                        "An Add-On experiment requires an Active "
```

Reply:
@peterbe After talking to @mythmon and Sunah yesterday we realized that we need to capture the experiment id to be passed to Normandy for monitoring and that it must be unique.

Reply:
The *code* doesn't doesn't do a check of uniqueness. It does a check of the `addon_experiment_id` being truthy. 

If it was something like:

```python
if Experiment.objects.filter(addon_experiment_id=addon_experiment_id).exclude(id=self.id).exists():
    raise ValueError('addon_experiment_id is not unique')
```
...the error message would match the code.

Reply:
Also, **uniqueness** is already protected by the constraint in postgres. Except it's `django.db.exceptions.IntegrityError` (me thinks) exception.

Reply:
Oh I see what you mean, yes that is true, we're only checking that it's set.  Django field validation checks the actual uniqueness, so if this case it's not failing a uniqueness validation it's just failing a presence check.  You're right I'll update the message to accurately reflect what is being checked.